File: src/test/java/io/github/binout/soccer/infrastructure/persistence/InMemoryPlayerRepository.java
package io.github.binout.soccer.infrastructure.persistence;

import io.github.binout.soccer.domain.player.Player;
import io.github.binout.soccer.domain.player.PlayerRepository;

import java.util.Comparator;
import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.stream.Stream;

public class InMemoryPlayerRepository implements PlayerRepository {

    private Map<String, Player> players;

    public InMemoryPlayerRepository() {
        players = new ConcurrentHashMap<>();
    }

    @Override
    public void add(Player player) {
        players.put(player.name(), player);
    }

    @Override
    public Stream<Player> all() {
        return players.values().stream().sorted(Comparator.comparing(Player::name));
    }

    @Override
    public Optional<Player> byName(String name) {
        return Optional.ofNullable(players.get(name));
    }
}

File: src/test/kotlin/io/github/binout/soccer/infrastructure/persistence/InMemoryPlayerRepository.kt
package io.github.binout.soccer.infrastructure.persistence

import io.github.binout.soccer.domain.player.Player
import io.github.binout.soccer.domain.player.PlayerRepository

import java.util.Comparator
import java.util.Optional
import java.util.concurrent.ConcurrentHashMap
import java.util.stream.Stream

class InMemoryPlayerRepository : PlayerRepository {

    private val players: MutableMap<String, Player> = ConcurrentHashMap()

    override fun add(player: Player) {
        players[player.name] = player
    }

    override fun all(): List<Player> = players.values.sortedBy { it.name }

    override fun byName(name: String): Player? = players[name]
}
